### PR TITLE
feat: implemented token and credential endpoints for openid4vc issuer 

### LINKED ISSUE #1942: [OpenID4CI] Implement mock issuer for pre-auth flow

### PR DESCRIPTION
**Pull request by birtony**
fixes #1942

https://user-images.githubusercontent.com/33902374/212208937-3a7cd92e-faf9-4fac-a652-ee3f1bbbdb1f.mov

Signed-off-by: Anton Biriukov <anton.biriukov@avast.com>

### CODE REVIEW

File: test/mock/adapter/adapter.go [hunk 1]
```diff
@@ -1127,15 +1119,192 @@ func (v *adapterApp) openid4vcInitiatePreAuthorizedIssuance(w http.ResponseWrite
...
-		url.QueryEscape(credType) + "&pre-authorized_code=" + url.QueryEscape(uuid.NewString()) + "&user_pin_required=true"
+		url.QueryEscape(credType) + "&pre-authorized_code=" + url.QueryEscape(preAuthCode) + "&user_pin_required=true"
 
 	err = t.Execute(w, openid4ciDemo{
 		InitiateUrl: initiateUrl,
 		Pin:         pinNumber,
 	})
 }
 
+func (v *adapterApp) openid4vcIssuerTokenEndpoint(w http.ResponseWriter, r *http.Request) {
```
Comment (sudeshrshetty) [inline test/mock/adapter/adapter.go:1150]: so this token endpoint supports only pre-authorized code flow?

File: test/mock/adapter/adapter.go [hunk 2]
```diff
@@ -1127,15 +1119,192 @@ func (v *adapterApp) openid4vcInitiatePreAuthorizedIssuance(w http.ResponseWrite
...
+	}
+
+	var authRequest map[string]string
+	err = json.Unmarshal(authRqstBytes, &authRequest)
+	if err != nil {
+		sendOIDCErrorResponse(w, "failed to read request", http.StatusInternalServerError)
+		return
+	}
+
+	if userPin != authRequest["pin"] {
```
Comment (sudeshrshetty) [inline test/mock/adapter/adapter.go:1177]: Actually `pin` is optional

**Review by sudeshrshetty — APPROVED** [review]

File: test/mock/adapter/adapter.go (hunk 1 above)
Comment (birtony) [reply test/mock/adapter/adapter.go:1150]: At this point - yes

File: test/mock/adapter/adapter.go (hunk 2 above)
Comment (birtony) [reply test/mock/adapter/adapter.go:1177]: Yes, but if the issuer sets `user_pin_required = true` in the initiate issuance response, then `pin` is required. We could make this more flexible by customizing the issuer, if needed

**Review by rolsonquadras — APPROVED** [review]

**Pull request merged**